core/cluster/process.go

```diff
@@ -0,0 +1,38 @@
+package cluster
```

Comment:
We could name the package `clustersync` and provide semantics similar to the `sync` package.
`sync.Once` for running once.
`sync.Mutex` for a mutual-exclusive lock.